pages/api/plants/[id]/index.js

```diff
@@ -0,0 +1,24 @@
+import mongoose from "mongoose";
+import Plant from "@/utils/db_plants";
+
+export default async function handler(req, res) {
+  if (req.method === "GET") {
+    if (!req.query.id) {
+      return res.status(400).json({ error: "Missing room ID" });
+    }
+
+    console.log("QUERY", req.query.id);
+
+    const plantsInRoom = await Plant.aggregate([
+      { $match: { room: new mongoose.Types.ObjectId(req.query.id) } },
+    ]);
```

Comment:
not sure, but the `aggregate` method might be a bit "take a sledgehammer to crack a nut". The `find` method might be a enough? 

That said, the `aggregate` method is a very nice solution though. Just a bit too complex for your programmer colleagues ;-)

Reply:
@mpagels Mareike actually suggested to use `aggregate`, because i already tried `find` and was stuck. Since it's finally working after 1,5 days of bug fixing, i'd like to leave it that way if that's okay 😭